array-filter/filter.js

```diff
@@ -0,0 +1,14 @@
+/*
+ define filter() from scratch as if it did not already exist on arrays.
+*/
+
+function filter(array, predicate) {
+  const newArray = [];
+
+  for (let i = 0; i < array.length; i++) {
+    if (predicate[array[i]] === true) {
+      newArray.push(array[i]);
+    }
+  }
+  return newArray;
+}
```

Comment:
You should actually try this function to see if it returns what you think it should.